### PR TITLE
Fix duplicate index handling and add test cases

### LINKED ISSUE #70: Add more test cases for DeleteCommandParser and DeleteCommand
Currently, DeleteCommandParser and DeleteCommand have limited test coverage.
- Edge cases such as invalid inputs, duplicate indices, and whitespace handling need to be tested.
- Additional test cases should ensure robustness and correct behavior.

### PR DESCRIPTION
**Pull request by nictjh**
Currently, DeleteCommandParser does not properly reject duplicate indices, leading to unexpected behavior when parsing user input.

This commit:
- Detects duplicate indices early and rejects them with an error.
- Adds test cases for empty input, duplicate indices, and negative indices.
- Adds test case for empty list delete.
- Fixes #70 , #72 

This improves parser robustness and prevents unintended command execution.

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -18,6 +18,7 @@ public class Messages {
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
     public static final String MESSAGE_DUPLICATE_FIELDS =
                 "Multiple values specified for the following single-valued field(s): ";
+    public static final String MESSAGE_DUPLICATE_INDEX = "Duplicate indices are not allowed for this command. %s";
```
Comment (YashvanGH) [inline src/main/java/seedu/address/logic/Messages.java:21]: Good thinking to separate out the message logic 👍

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -52,10 +52,11 @@ public CommandResult execute(Model model) throws CommandException {
...
 
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS,
-                personsToDelete.stream()
-                        .map(Messages::format)
-                        .collect(Collectors.joining(","))));
+        String deletedPersonsSummary = personsToDelete.stream()
+                .map(Messages::format)
+                .collect(Collectors.joining(","));
+
+        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, deletedPersonsSummary));
```
Comment (YashvanGH) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:59]: I think the code is clearer by separating out the logic 👍

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java [hunk 3]
```diff
@@ -25,6 +27,13 @@ public class DeleteCommandParserTest {
 
     private DeleteCommandParser parser = new DeleteCommandParser();
 
+    @Test
```
Comment (YashvanGH) [inline src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java:30]: Good coverage on new tests! 👍

**Review by YashvanGH — COMMENTED** [review]
Overall LGTM. Honestly Im not too sure how you can do the coverage on the new DuplicateIndexParseException. My (not so professional) advice would be to see how other classes wrote the tests for individual exceptions.
